### ISSUE TITLE
loop control statements allowed in while conditions

### ISSUE THREAD

**Issue by ridiculousfish** [fish-shell/fish-shell#1212]
This silliness is allowed:

```
while break
    echo Unreachable
end

while continue
    echo Loops forever
end
```

**Comment by zanchey** [comment]
```
while continue
    echo Loops forever
end
```

returns immediately (as of 665314b240a9).

**Comment by faho** [comment]
Retargeting this for later - the worst issue (endless looping with continue) has been fixed, so this isn't critical.

**Comment by ridiculousfish** [comment]
This coincidentally got fixed along with #4820
